src/Explorer/Controls/ThroughputInput/ThroughputInputReactComponentAutoPilotV3.tsx

```diff
@@ -0,0 +1,249 @@
+  maximum: number;
+  step?: number;
+  isEnabled?: boolean;
+  costsVisible: boolean;
+  requestUnitsUsageCost: JSX.Element;
+  spendAckChecked?: boolean;
+  spendAckId?: string;
+  spendAckText?: string;
+  spendAckVisible?: boolean;
+  showAsMandatory?: boolean;
+  isFixed?: boolean;
+  label: string;
+  infoBubbleText?: string;
+  canExceedMaximumValue?: boolean;
+  cssClass?: string;
+  setAutoPilotSelected: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  isAutoPilotSelected: boolean;
+  throughputAutoPilotRadioId: string;
+  throughputProvisionedRadioId: string;
+  throughputModeRadioName: string;
+  autoPilotUsageCost: JSX.Element;
+  showAutoPilot?: boolean;
+  overrideWithAutoPilotSettings: boolean;
+  overrideWithProvisionedThroughputSettings: boolean;
+  maxAutoPilotThroughput: StatefulValue<number>;
+  setMaxAutoPilotThroughput: (newThroughput: number) => void;
+}
+
+interface ThroughputInputState {
+  spendAckChecked: boolean;
+}
+
+export class ThroughputInputAutoPilotV3Component extends React.Component<
+  ThroughputInputAutoPilotV3Props,
+  ThroughputInputState
+> {
+  private static readonly defaultStep = 100;
+  private static readonly zeroThroughput = 0;
+  private cssClass: string;
+  private step: number;
+
+  public constructor(props: ThroughputInputAutoPilotV3Props) {
+    super(props);
+    this.state = {
+      spendAckChecked: this.props.spendAckChecked
+    };
+
+    this.step = this.props.step ?? ThroughputInputAutoPilotV3Component.defaultStep;
+    this.cssClass = this.props.cssClass || "textfontclr collid migration";
+  }
+
+  private onAutoPilotThroughputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    let newThroughput = parseInt(e.currentTarget.value);
+    newThroughput = isNaN(newThroughput) ? ThroughputInputAutoPilotV3Component.zeroThroughput : newThroughput;
+    this.props.setMaxAutoPilotThroughput(newThroughput);
+  };
+
+  private onThroughputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    let newThroughput = parseInt(e.currentTarget.value);
+    newThroughput = isNaN(newThroughput) ? ThroughputInputAutoPilotV3Component.zeroThroughput : newThroughput;
+
+    if (this.props.overrideWithAutoPilotSettings) {
+      this.props.setMaxAutoPilotThroughput(newThroughput);
+    } else {
+      this.props.setThroughput(newThroughput);
+    }
+  };
+
+  private getThroughputValue = (): string => {
+    if (this.props.throughput.current) {
+      if (this.props.throughput.current === 0) {
+        return "";
+      } else {
+        return this.props.throughput.current.toString();
+      }
+    }
+    return "";
+  };
+
+  private getMaxAutoPilotThroughputValue = (): string => {
+    if (this.props.overrideWithProvisionedThroughputSettings) {
+      return "";
+    } else {
+      return this.props.maxAutoPilotThroughput.current === 0
+        ? ""
+        : this.props.maxAutoPilotThroughput.current.toString();
+    }
+  };
+
+  public render(): JSX.Element {
```

Comment:
Break this function into sub-functions that are easier to unit test. Like in Settings, the visible/non visible (as in `flag && <></>`) code paths are probably natural boundaries to break it down. If you think that any of the sub render part is reusable elsewhere, make it a separate component.